src/Parser/DefaultParser.php

```diff
@@ -27,13 +28,13 @@ public function __construct(Lexer $lexer = null)
     public function parse(string $expression) : Ast\Node
     {
         $this->expression = $expression;
-        $this->pointer    = 0;
         $this->tokens     = $this->lexer->tokenize($this->expression);
+        $this->pointer    = 0;
+        $this->pointerMax = count($this->tokens) - 1;
 
         // Check for an empty expression explicitly.
-        // This is not technically necessary, but allows for a more helpful error message over the more
-        // generic "Expected 'IDENTIFIER', got 'EOF'." that would otherwise be produced.
-        if ($this->token()->is(Token::EOF)) {
+        // This allows for a nicer error message than would otherwise be produced by parseExpression().
+        if ($expression === '') {
```

Comment:
s: Should we trim the expression before this check as well?

Reply:
Yeah good point. Strictly we should just check for the `EOF` token because the `Lexer` has already discarded the whitespace. This is how it was before. I'll revert to that.